include/fmt/chrono.h

```diff
@@ -759,18 +777,23 @@ inline std::chrono::duration<Rep, std::milli> get_milliseconds(
   return std::chrono::duration<Rep, std::milli>(static_cast<Rep>(ms));
 }
 
-template <typename Rep, typename OutputIt>
+template <typename Char, typename Rep, typename OutputIt>
 OutputIt format_duration_value(OutputIt out, Rep val, int precision) {
-  if (precision >= 0) return format_to(out, "{:.{}f}", val, precision);
-  return format_to(out, std::is_floating_point<Rep>::value ? "{:g}" : "{}",
-                   val);
+  const Char pr_f[] { '{', ':', '.', '{', '}', 'f', '}', 0 };
+  if (precision >= 0) return format_to(out, pr_f, val, precision);
+  const Char fp_f[] { '{', ':', 'g', '}', 0 };
+  const Char format[] { '{', '}', 0 };
+  return format_to(out, std::is_floating_point<Rep>::value ? fp_f : format, val);
 }
 
-template <typename Period, typename OutputIt>
+template <typename Char, typename Period, typename OutputIt>
 OutputIt format_duration_unit(OutputIt out) {
-  if (const char* unit = get_units<Period>()) return format_to(out, "{}", unit);
-  if (Period::den == 1) return format_to(out, "[{}]s", Period::num);
-  return format_to(out, "[{}/{}]s", Period::num, Period::den);
+  const Char format[] { '{', '}', 0 };
+  if (const Char* unit = get_units<Char, Period>()) return format_to(out, format, unit);
```

Comment:
Correction: [`utf8_to_utf16`](https://github.com/fmtlib/fmt/blob/09a13244c85de35471ff7a660b17360111b14430/include/fmt/format.h#L948) + copy because of micro.

Reply:
Still need a way to determine if it should just do `copy` if `Char` is narrow, instead `utf8_to_utf16` then `copy`. How would you like me to approach this? Is `if FMT_CONSTEXPR (std::is_same<Char, wchar_t>::value)` allowed?

Reply:
I went with the `if constexpr` solution, since it was the most compact way I could think of. It seems to work well. Let me know if it is satisfactory.